services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java

```diff
@@ -0,0 +1,278 @@
+import software.amazon.awssdk.services.cloudfront.model.CannedSignerRequest;
+import software.amazon.awssdk.services.cloudfront.model.CustomSignerRequest;
+import software.amazon.awssdk.services.cloudfront.url.SignedUrl;
+
+/**
+ *
+ * Utilities for working with CloudFront distributions
+ * <p>
+ *     To securely serve private content by using CloudFront, you can require that users access your private content by using
+ *     special CloudFront signed URLs or signed cookies. You then develop your application either to create and distribute signed
+ *     URLs to authenticated users or to send Set-Cookie headers that set signed cookies for authenticated users.
+ * </p>
+ * <p>
+ *     Signed URLs take precedence over signed cookies. If you use both signed URLs and signed cookies to control access to the
+ *     same files and a viewer uses a signed URL to request a file, CloudFront determines whether to return the file to the
+ *     viewer based only on the signed URL.
+ * </p>
+ *
+ */
+@Immutable
+@ThreadSafe
+@SdkPublicApi
+public final class CloudFrontUtilities {
+
+    private static final String KEY_PAIR_ID_KEY = "CloudFront-Key-Pair-Id";
+    private static final String SIGNATURE_KEY = "CloudFront-Signature";
+    private static final String EXPIRES_KEY = "CloudFront-Expires";
+    private static final String POLICY_KEY = "CloudFront-Policy";
+
+    private CloudFrontUtilities() {
+    }
+
+    public static CloudFrontUtilities create() {
+        return new CloudFrontUtilities();
+    }
+
+    /**
+     * Returns a signed URL with a canned policy that grants universal access to
+     * private content until a given date.
+     * For more information, see <a href=
+     * "https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-creating-signed-url-canned-policy.html"
+     * >Creating a signed URL using a canned policy</a>.
+     *
+     * <p>
+     *     This is a convenience which creates an instance of the {@link CannedSignerRequest.Builder} avoiding the need to
+     *     create one manually via {@link CannedSignerRequest#builder()}
+     * </p>
+     *
+     * @param request A {@link Consumer} that will call methods on {@link CannedSignerRequest.Builder} to create a request.
+     * @return A signed URL that will permit access to a specific distribution
+     *         and S3 object.
+     */
+    public SignedUrl getSignedUrlWithCannedPolicy(Consumer<CannedSignerRequest.Builder> request) {
+        return getSignedUrlWithCannedPolicy(CannedSignerRequest.builder().applyMutation(request).build());
+    }
+
+    /**
+     * Returns a signed URL with a canned policy that grants universal access to
+     * private content until a given date.
+     * For more information, see <a href=
+     * "https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-creating-signed-url-canned-policy.html"
+     * >Creating a signed URL using a canned policy</a>.
+     *
```

Comment:
Can we add some code snippet for all public APIs?

> Use @snippet to add code snippets.

https://github.com/aws/aws-sdk-java-v2/blob/master/docs/design/APIReference.md#style-guideline

Reply:
to add inline code snippets